test/end-to-end/pages/main.js

```diff
@@ -6,6 +6,6 @@ module.exports = class MainPage {
     this.mailUrl = pageConfig.root;
     this.title = title;
     this.iframeSelector = '.container-frame';
-    this.iframeLoadingTime = 3000;
+    this.iframeLoadingTime = 500;
```

Comment:
This seems like making it unnecessary tight for loaded test machines?

Reply:
I know it's a ugly hardcode time wait. I have no idea how to avoid this. Before enter into a iframe, the page has to wait the iframe loaded. I tried lots of `wait(<page element>)`, it does not work unfortunately. Do you have any suggestion on this? Thanks.

Reply:
I wasn't complaining about the fact that it's hardcoded, just that this change seems to make the timeout too short. On loaded infrastructure 3s seems more robust to me than 0.5s.

Reply:
Thanks for your suggestion.
The first reason is I try to speed up test duration. This wait does not need to wait the whole page loaded, just make sure iframe loaded, not page inside iframe. I've tried 3, 2, 1 and 0.5s, all of them work.
Another reason to short this wait time is it has to be shorter than default wait time. I setup the default wait time to 3s, so this wait time has to be shorter than 3s.
I'll make more test on this to keep test stable with a shorter timer.

Reply:
@henrywang : Ah, thanks for the explanation. I misunderstood this then. This sounds rather brittle - chromium and the DevTools protocol can tell you when a page or an iframe is fully loaded, so there is no reason to have to use a static timeout. However, this isn't making things worse, so fine for me for now.

Reply:
@martinpitt, thanks for your reply. We can talk DevTools protocol offline.